File: winter-compiler/src/test/resources/InjectConstructorWithProviderAndLazyArguments_WinterFactory.java
package test;

import io.jentz.winter.ClassTypeKey;
import io.jentz.winter.Component;
import io.jentz.winter.GenericClassTypeKey;
import io.jentz.winter.Graph;
import io.jentz.winter.TypeKey;
import io.jentz.winter.inject.Factory;
import io.jentz.winter.inject.InterOp;
import java.lang.Override;
import java.lang.String;
import java.util.List;
import javax.annotation.Generated;
import kotlin.LazyKt;

@Generated(
        value = "io.jentz.winter.compiler.WinterProcessor",
        date = "2019-02-10T14:52Z"
)
public final class InjectConstructorWithProviderAndLazyArguments_WinterFactory implements Factory<InjectConstructorWithProviderAndLazyArguments> {
    @Override
    public InjectConstructorWithProviderAndLazyArguments invoke(final Graph graph) {
        return new InjectConstructorWithProviderAndLazyArguments(
                () -> graph.instanceByKey(new ClassTypeKey<String>(String.class, "string")),
                LazyKt.lazy(() -> graph.instanceOrNullByKey(new GenericClassTypeKey<List<String>>("stringList") {}))
        );
    }

    @Override
    public TypeKey<InjectConstructorWithProviderAndLazyArguments> register(
            final Component.Builder builder, final boolean override) {
        TypeKey<test.InjectConstructorWithProviderAndLazyArguments> key = new ClassTypeKey<InjectConstructorWithProviderAndLazyArguments>(InjectConstructorWithProviderAndLazyArguments.class, null);
        InterOp.prototype(builder, key, override, this);
        return key;
    }
}

File: winter-compiler/src/test/resources/InjectConstructorWithProviderAndLazyArguments_WinterFactory.kt
package test

import io.jentz.winter.Component
import io.jentz.winter.Graph
import io.jentz.winter.TypeKey
import io.jentz.winter.inject.Factory
import javax.annotation.Generated
import kotlin.Boolean
import kotlin.String
import kotlin.collections.List

@Generated(
  value = ["io.jentz.winter.compiler.WinterProcessor"],
  date = "2019-02-10T14:52Z"
)
class InjectConstructorWithProviderAndLazyArguments_WinterFactory :
    Factory<InjectConstructorWithProviderAndLazyArguments> {
  override fun invoke(graph: Graph): InjectConstructorWithProviderAndLazyArguments =
      InjectConstructorWithProviderAndLazyArguments(
      { graph.instance<String>(qualifier = "string") },
      lazy { graph.instanceOrNull<List<String>>(qualifier = "stringList", generics = true) }
  )

  override fun register(builder: Component.Builder, override: Boolean):
      TypeKey<InjectConstructorWithProviderAndLazyArguments> = builder.prototype(override =
      override, factory = this)
}
